src/aleph_client/commands/instance/__init__.py

```diff
@@ -353,57 +355,137 @@ async def delete(
         typer.echo(f"Instance {item_hash} has been deleted. It will be removed by the scheduler in a few minutes.")
 
 
-async def _get_ipv6_address(message: InstanceMessage, node_list: NodeInfo) -> Tuple[str, str]:
+async def _get_instance_details(message: InstanceMessage, node_list: NodeInfo) -> Tuple[str, str]:
     async with ClientSession() as session:
+        hold = not message.content.payment or message.content.payment.type == PaymentType["hold"]
+        confidential = (
+            hasattr(message.content, "environment")
+            and hasattr(message.content.environment, "trusted_execution")
+            and hasattr(message.content.environment.trusted_execution, "firmware")
+            and len(message.content.environment.trusted_execution.firmware) == 64
+        )
+        details = dict(
+            ipv6_logs=None,
+            payment="hold\t   " if hold else message.content.payment.type,
+            confidential=confidential,
+        )
         try:
-            if not message.content.payment:
-                # Fetch from the scheduler API directly if no payment
+            if hold and not confidential:
+                # Fetch from the scheduler API directly if no payment or no receiver
                 status = await fetch_json(
                     session,
                     f"https://scheduler.api.aleph.cloud/api/v0/allocation/{message.item_hash}",
                 )
-                return status["vm_hash"], status["vm_ipv6"]
+                details["ipv6_logs"] = status["vm_ipv6"]
+                return status["vm_hash"], details
             for node in node_list.nodes:
-                if node["stream_reward"] == message.content.payment.receiver:
-
+                if node["stream_reward"] == message.content.payment.receiver or (
+                    hasattr(message.content, "requirements")
+                    and hasattr(message.content.requirements, "node")
+                    and node["hash"] == message.content.requirements.node.node_hash
+                ):
                     # Handle both cases where the address might or might not end with a '/'
-                    path: str = (
-                        f"{node['address']}about/executions/list"
-                        if node["address"][-1] == "/"
-                        else f"{node['address']}/about/executions/list"
-                    )
+                    path = f"{node['address'].rstrip('/')}/about/executions/list"
                     # Fetch from the CRN API if payment
                     executions = await fetch_json(session, path)
                     if message.item_hash in executions:
-                        ipv6_address = executions[message.item_hash]["networking"]["ipv6"]
-                        return message.item_hash, ipv6_address
-
-            return message.item_hash, "Not available (yet)"
+                        details["ipv6_logs"] = executions[message.item_hash]["networking"]["ipv6"].split("/")[0]
```

Comment:
Would this make it clearer than the split on string ? (import `ipaddress` from builtins)

```suggestion
                        interface = ipaddress.IPv6Interface(executions[message.item_hash]["networking"]["ipv6"]
                        details["ipv6_logs"] = str(interface).ip)
```